contracts/vePERPRewardDistributor.sol

```diff
@@ -0,0 +1,199 @@
+    //**********************************************************//
+
+    //
+    // MODIFIER
+    //
+
+    /// @notice Modifier to check if the caller's vePERP lock time is over minLockDuration
+    modifier userLockTimeCheck(address user) {
+        uint256 currentEpochStartTimestamp = (block.timestamp / _WEEK) * _WEEK; // round down to the start of the epoch
+        uint256 userLockEndTimestamp = IvePERP(_vePERP).locked__end(user);
+
+        // vePRD_LTM: vePERP lock time is less than minLockDuration
+        require(userLockEndTimestamp >= currentEpochStartTimestamp + _minLockDuration, "vePRD_LTM");
+        _;
+    }
+
+    //
+    // ONLY OWNER
+    //
+
+    function initialize(
+        address tokenArg,
+        address vePERPArg,
+        uint256 minLockDurationArg
+    ) external initializer {
+        // vePRD_TNC: token is not a contract
+        require(tokenArg.isContract(), "vePRD_TNC");
+
+        __MerkleRedeem_init(tokenArg);
+
+        setVePERP(vePERPArg);
+        setMinLockDuration(minLockDurationArg);
+
+        // approve the vePERP contract to spend the PERP token
+        token.approve(vePERPArg, uint256(-1));
+    }
+
+    function seedAllocations(
+        uint256 week,
+        bytes32 merkleRoot,
+        uint256 totalAllocation
+    ) public override onlyOwner {
+        // vePRD_TIZ: total allocation is zero
+        require(totalAllocation > 0, "vePRD_TIZ");
+        super.seedAllocations(week, merkleRoot, totalAllocation);
+        merkleRootIndexes.push(week);
+        emit AllocationSeeded(week, totalAllocation);
+    }
+
+    /// @dev In case of vePERP migration, unclaimed PERP would be able to be deposited to the new contract instead
+    function setVePERP(address vePERPArg) public onlyOwner {
+        // vePRD_vePNC: vePERP is not a contract
+        require(vePERPArg.isContract(), "vePRD_vePNC");
+        emit VePERPChanged(_vePERP, vePERPArg);
+        _vePERP = vePERPArg;
+    }
+
+    function setMinLockDuration(uint256 minLockDurationArg) public onlyOwner {
+        emit MinLockDurationChanged(_minLockDuration, minLockDurationArg);
+        _minLockDuration = minLockDurationArg;
+    }
+
+    //
+    // PUBLIC NON-VIEW
+    //
+
+    /// @notice Claim vePERP reward for a week
+    /// @dev Overwrite the parent's function because vePERP distributor doesn't follow the inherited behaviors
+    ///      from its parent. More specifically, it uses deposit_for() instead of transfer() to distribute the rewards.
+    /// @param liquidityProvider Liquidity provider address
+    /// @param week Week number of the reward claimed
+    /// @param claimedBalance Amount of vePERP reward claimed
+    /// @param merkleProof Merkle proof of the week's allocation
+    function claimWeek(
+        address liquidityProvider,
+        uint256 week,
+        uint256 claimedBalance,
+        bytes32[] calldata merkleProof
+    ) public override userLockTimeCheck(liquidityProvider) {
```

Comment:
Internal audit:
Why do we need the modifier userLockTimeCheck?
I saw there is [lock-time checking](https://github.com/perpetual-protocol/voting-escrow/blob/main/contracts/vePERP.vy#L473) in `deposit_for` as well. What't the difference between these two?